test/index.spec.ts

```diff
@@ -0,0 +1,30 @@
+import axios from 'axios';
+import { ApolloServer } from 'apollo-server';
+import { AppDataSource } from '../src/data-source';
+import { expect } from 'chai';
+import { resolvers } from '../src/graphql/resolvers';
+import { typeDefs } from '../src/graphql/schema';
+
+describe('query hello', function () {
+  before('Starting server', function (done) {
+    AppDataSource.initialize()
+      .then(() => {
+        const server = new ApolloServer({ typeDefs, resolvers });
+        server.listen().then(({ url }) => {
+          console.log(`Server listening at ${url}`);
+        });
+      })
+      .catch((error) => console.log(error))
+      .then(done);
+  });
```

Comment:
A sintaxe do async await geralmente é menos aninhada e mais fácil de compreender:

```ts
before('Starting Server', async () => {
  await AppDataSource.initialize();
  const server = new ApolloServer({ typeDefs, resolvers });
  const { url } = await new ApolloServer({ typedefs, resolvers });
  console.log(`Server listening at ${url}`);
});
```

Melhor né? Assim nem precisa do "done" do mocha. Quanto ao erro, vc até poderia pegar com try-catch, mas como estava implementado, ele vai acabar seguindo com os testes mesmo em caso de erro, mas se der algum erro com o setup do server nem faz sentido executar os testes, certo?

Reply:
Fiz essa alteração no commit "Change then/catch to async/await"